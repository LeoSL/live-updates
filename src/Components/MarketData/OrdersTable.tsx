import {
  Table,
  TableBody,
  TableCell,
  TableRow,
  makeStyles,
  Typography,
} from "@material-ui/core";
import { ComponentType } from "react";
import { Orderbook } from ".";

import { addOrdersType, evenSmarterShuffledArray } from "../../lib/utils";

const RED = "#e5210e15";
const RED_TEXT = "#e5210e";
const GREEN = "#00923915";
const GREEN_TEXT = "#009239";

const useStyles = makeStyles({
  root: {
    display: "flex",
    justifyContent: "center",
    backgroundColor: "white",
    width: "100%",
  },
  priceCell: {
    "& p:hover": {
      fontWeight: "bold",
      cursor: "pointer",
    },
  },
});

type MarketDataRowProps = {
  type: string;
  price: string;
};

const MarketDataRow: ComponentType<MarketDataRowProps> = ({ type, price }) => {
  const classes = useStyles();

  return (
    <TableRow>
      <TableCell
        key={price}
        style={{
          width: "33vw",
          padding: "3px 10px",
          color: `${type === "asks" ? GREEN_TEXT : RED_TEXT}`,
          backgroundColor: `${type === "asks" ? GREEN : RED}`,
        }}
        align={type === "asks" ? "right" : "left"}
      >
        <div className={classes.priceCell}>
          <Typography>{price}</Typography>
        </div>
      </TableCell>
    </TableRow>
  );
};

export const OrdersTable: ComponentType<Orderbook> = ({ asks, bids }) => {
  const classes = useStyles();
  const orders = addOrdersType("asks", asks).concat(
    addOrdersType("bids", bids)
  );

  return (
    <Table cellPadding={0} className={classes.root}>
      <TableBody>
        {evenSmarterShuffledArray(orders).map((props: any) => (
          <MarketDataRow key={Math.random()} {...props} />
        ))}
      </TableBody>
    </Table>
  );
};
