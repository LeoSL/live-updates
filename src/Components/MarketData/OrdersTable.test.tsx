import { render } from "@testing-library/react";
import { OrdersTable } from "./OrdersTable";

it("renders correctly", () => {
  const props = {
    asks: [{ price: "12000" }],
    bids: [{ price: "20000" }],
  };
  const { container } = render(<OrdersTable {...props} />);
  expect(container).toMatchSnapshot();
});
