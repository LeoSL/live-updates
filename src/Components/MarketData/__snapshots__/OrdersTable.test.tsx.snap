// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    cellpadding="0"
    class="MuiTable-root makeStyles-root-1"
  >
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          style="width: 33vw; padding: 3px 10px; color: rgb(229, 33, 14); background-color: rgba(229, 33, 14, 0.082);"
        >
          <div
            class="makeStyles-priceCell-2"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              20000
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          style="width: 33vw; padding: 3px 10px; color: rgb(0, 146, 57); background-color: rgba(0, 146, 57, 0.082);"
        >
          <div
            class="makeStyles-priceCell-2"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              12000
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
